### PR TITLE
feat: better default for the php.ini path

### PR DESCRIPTION
**Pull request by dunglas**
## What does this PR do?

By default, the `php.ini` path is `/lib/php.ini`. This value is "hardcoded" in PHP but can be changed with configure flags. Most (all?) Linux distributions as well as the official PHP Docker images change this default path to `/usr/local/etc/php/php.ini` or `/etc/php.ini`.

This default value is very confusing for end users (see for instance https://github.com/dunglas/frankenphp/issues/1066, it's a recurrent issue we have with FrankenPHP).

Additionally, `/lib` doesn't even exist by default on Mac, and cannot be created because because the system filesystem is read-only.

This PR aligns the default path to the one used by official Docker images: `/usr/local/etc/php/php.ini`
In addition, it loads all files defined in `/usr/local/etc/php/conf.d`, as the Docker images do.

It can still be changed easily by end users if needed by overriding `SPC_CMD_PREFIX_PHP_CONFIGURE`.

### CODE REVIEW

**Comment by crazywhalecc** [comment]
> It can still be changed easily by end users if needed by overriding SPC_CMD_PREFIX_PHP_CONFIGURE.

Do you think this happens often? I just found out there is a PR about this before switching to `env.ini`: #511, which adds command line arguments to specify. I'm not sure which way is more suitable for most people, but we can change the default anyway.

**Comment by dunglas** [comment]
It could be convenient to provide an easier way to change the path, indeed. Changing the default and adding a new option looks like the best option to me.

**Comment by dunglas** [comment]
@crazywhalecc great! How would you like to proceed? Change the default value in the option definition directly and add a new option to set `--with-config-file-scan-dir`?

**Comment by crazywhalecc** [comment]
> @crazywhalecc great! How would you like to proceed? Change the default value in the option definition directly and add a new option to set `--with-config-file-scan-dir`?

Yes (only if we use command options).

**Comment by dunglas** [comment]
Code and docs updated.

I used DeepL for the Chinese translation, it may be totally wrong!

**Review by crazywhalecc — COMMENTED** [review]
Thanks for your contribution. I will adjust the documentation section after finish modifications.

I also found that it seems that I missed Windows support, and Windows configure does not seem to be able to modify the default `config-file-path`, only `config-file-scan-dir`. And Windows cli will look for php.ini in  `C:\Windows\` and `$(cwd)`.

File: src/SPC/command/BuildCliCommand.php [hunk 1]
```diff
@@ -32,7 +34,8 @@ public function configure(): void
         $this->addOption('no-strip', null, null, 'build without strip, in order to debug and load external extensions');
         $this->addOption('enable-zts', null, null, 'enable ZTS support');
         $this->addOption('disable-opcache-jit', null, null, 'disable opcache jit');
-        $this->addOption('with-config-file-path', null, InputOption::VALUE_REQUIRED, 'Set the path in which to look for php.ini');
+        $this->addOption('with-config-file-path', null, InputOption::VALUE_OPTIONAL, 'Set the path in which to look for php.ini', $isWindows ? null : '/usr/local/etc/php');
+        $this->addOption('with-config-file-scan-dir', null, InputOption::VALUE_OPTIONAL, 'Set the directory to scan for .ini files after reading php.ini', $isWindows ? null : '/usr/local/etc/php/conf.d');
```
[the remark this answers is not in the document]
Comment (dunglas) [reply src/SPC/command/BuildCliCommand.php:38]: Indeed... fixed.

(my intention was to allow passing `null` but it's not possible and probably not useful)

**Pull request merged**